src/EnrollUserController.js

```diff
@@ -0,0 +1,57 @@
+/*!
+ * Copyright (c) 2019, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+define([
+  'okta',
+  'models/EnrollUser',
+  'util/BaseLoginController',
+  'views/enrollUser/EnrollUserForm',
+  'views/shared/FooterWithBackLink',
+  'q'
+],
+function (
+  Okta,
+  EnrollUser,
+  BaseLoginController,
+  EnrollUserForm,
+  FooterWithBackLink,
+  Q
+) {
+  var _ = Okta._;
+  return BaseLoginController.extend({
+    className: 'enroll-user',
+    initialize: function (options) {
+      this.options = options || {};
+      // create model
+      this.model = new EnrollUser(this.options);
+    },
+    fetchInitialData: function () {
+      // If user is unauthenticated and starts enroll flow make a post call to transition state to PROFILE_REQUIRED
+      if (this.options.appState.get('isUnauthenticated')) {
+        return this.model.getEnrollFormData()
+          .then(_.bind(function () {
+            this.renderForm();
```

Comment:
I believe `BaseLoginController` will do `controller.render` after `fetchInitialData` succeeds, so maybe you don't have to be explicit about rendering the form here, and you just have to return the promise... you could give it a try

Reply:
@mauriciocastillosilva-okta I was able to get it working by moving the rendering logic the logic inside `postRender` ?
```
    fetchInitialData: function () {
      // If user is unauthenticated and starts enroll flow make a post call to transition state to PROFILE_REQUIRED
      if (this.options.appState.get('isUnauthenticated')) {
        return this.model.getEnrollFormData();
      } else {
        var deferred = Q.defer();
        deferred.resolve(this.options.appState.get('lastAuthResponse'));
        return deferred.promise;
      }
    },
    postRender: function () {
      var form = new EnrollUserForm(this.toJSON());
      this.add(form);
      this.add(new FooterWithBackLink(this.toJSON()));
      this.addListeners();
    }
```

Reply:
After debugging a little more, I found that i'll have to use `trapAuthResponse` when `isProfileRequired` is true.  When user clicks sign up in an `UNAUTHENTICATED` state, it switches the route to `signin/enroll-user`, then we do the fetch to get `PROFILE_REQUIRED` form data and eventually do a `setTransaction(trans)`, which calls routerUtil and tries to change the route to `signin/enroll-user` again.